component/component.go

```diff
@@ -112,15 +146,26 @@ type Factory interface {
 	// Type gets the type of the component created by this factory.
 	Type() config.Type
 
+	// StabilityLevel gets the stability level of the component.
+	StabilityLevel(config.Type) StabilityLevel
```

Comment:
Is this "config.DataType"? I know is the same, but this is what I think makes sense.

Reply:
Yup 👍🏻